File: src/main/java/indi/fcwyzzr/minecraft/food_talks/common/mixin/accessors/mechanic/MobEffectInstanceAccessor.java
package indi.fcwyzzr.minecraft.food_talks.common.mixin.accessors.mechanic;

import net.minecraft.world.effect.MobEffectInstance;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

@Mixin(MobEffectInstance.class)
public interface MobEffectInstanceAccessor extends Comparable<MobEffectInstance> {
    
    @Accessor(value = "hiddenEffect")
    MobEffectInstance getHiddenEffect();
    
    @Accessor(value = "duration")
    void setDuration(int value);
}

File: src/main/kotlin/indi/muxin/food_talks/common/mixin/mechanic/MobEffectInstanceAccessor.kt
package indi.muxin.food_talks.common.mixin.mechanic

import net.minecraft.world.effect.MobEffectInstance
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.gen.Accessor

@Mixin(MobEffectInstance::class)
interface MobEffectInstanceAccessor : Comparable<MobEffectInstance> {
    @get:Accessor(value = "hiddenEffect")
    val hiddenEffect: MobEffectInstance?

    @Accessor(value = "duration")
    fun setDuration(value: Int)
}